### ISSUE TITLE
no nalgebra/std by default can lead to developer confusion

### ISSUE THREAD

**Issue by xd009642** [rust-cv/nshare#8]
So just had something at work when an engineer added nshare to move types between some existing nalgebra and ndarray code. And they got this error and couldn't figure it out:

```
error[E0599]: no method named `into_ndarray2` found for struct `Matrix<f64, Dynamic, Dynamic, VecStorage<f64, Dynamic, Dynamic>>` in the current scope
  --> matrix-thingy/src/foo.rs:23:27
   |
23 |     let a = b.into_ndarray2();
   |               ^^^^^^^^^^^^^ method not found in `Matrix<f64, Dynamic, Dynamic, VecStorage<f64, Dynamic, Dynamic>>`
```

They then copied in the example from the docs and it did the same, and then they asked me. I suggested turning on the std feature as it's absence may be leading to two nalgebra versions in the build and the trait only implemented on one of them and this solved the issue.

So, I'd propose making std a default feature. Anyone building for a no_std environment vets their dependencies and selects the no_std compatible features. Whereas everyone just assumes things will work _as expected_ on std envs. Also, the docs could be improved on this point (and should be even if we don't wish to make std default).

**Comment by vadixidav** [comment]
So `nalgebra` internally has independent types depending on whether it is std or not? Or is it that there are actually two nalgebras? I haven't encountered that before, but that is very concerning. I will look into this a bit more to see what might be going on. If we need to, we can make the change you proposed. We need to update to the latest nalgebra as well anyways.

**Comment by xd009642** [comment]
I haven't confirmed if it has, but it was my guess and it compiled after changing it. I haven't looked into it further than this :sweat_smile:

**Comment by vadixidav** [comment]
I looked into this more, and its probably because `nshare` itself only implements conversions to ndarray for these cases when the std feature is enabled here: https://github.com/rust-cv/nshare/blob/master/src/tondarray/nalgebra_impl.rs#L267.

I think this is a fair point. Lets make `std` a default feature.

**Comment by vadixidav** [comment]
Okay, so I ran into a problem. If I want `nshare` to not have any conversions by default, it wont be possible to have a feature that allows nalgebra's std components to be used by default because then that feature would bring nalgebra in. For instance:
```toml
[features]
default = ["nalgebra_std"]
nalgebra_std = ["nalgebra/std"]
```

This would bring in nalgebra's `std` feature, but it also brings in `nalgebra` in its entirety.

Though it pains me somewhat, perhaps the best thing to do here is to actually just bring in all of the crates by default. Just bringing in `nalgebra` by default would be strange, but bringing in all of them by default might just be what we should do.

I published version 0.8.0 with this change. If we don't like it, it can be changed back, so I will leave this issue open for now. The new version also has an updated nalgebra.
